modules/library/coverage/src/main/java/org/geotools/coverage/processing/OperationJAI.java

```diff
@@ -1079,4 +1099,86 @@ final RenderedImage getSource() {
             return null;
         }
     }
+    
+    protected void handleJAIEXTParams(ParameterBlockJAI parameters, ParameterValueGroup parameters2) {
+        return;
+    }
+    
+    protected static Map<String, Object> handleROINoDataProperties(Map<String, Object> properties,
+            ParameterBlockJAI parameters, GridCoverage2D sourceCoverage, String operationName,
+            int roiIndex, int noDataIndex, int backgroundIndex) {
+        Map<String, Object> prop = new HashMap<>();
+        if (properties != null) {
+            prop.putAll(properties);
+        }
+        // Setting the internal properties
+        if (JAIExt.isJAIExtOperation(operationName)) {
+            ROI roiParam = (ROI) parameters.getObjectParameter(roiIndex);
+            CoverageUtilities.setROIProperty(properties, roiParam);
+
+            Range noDataParam = (Range) parameters.getObjectParameter(noDataIndex);
+            if (noDataParam != null || roiParam != null) {
```

Comment:
Not so evident why the code does what it does inside this if, I'd add an explanation in a new comment